hudi-common/src/main/java/org/apache/hudi/common/config/ConfigOption.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.config;
+
+import java.util.Objects;
+
+/**
+ * A ConfigOption contains the key and an optional default value.
+ *
+ * @param <T> The type of the default value.
+ */
+public class ConfigOption<T> {
```

Comment:
if we are reusing this from Flink directly, we can add a mention in NOTICE?

Reply:
No we don't directly reuse its code so we don't need NOTICE here.